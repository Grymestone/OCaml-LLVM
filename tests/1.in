int do_math(int a) {
  int x = a * 5 + 3
}

double do_math2(double b) {
  double x = b * 10.0
}

int main() {
    double q = do_math2(2.0)
    int z = do_math(10)
    printf(z)
    printf(q)
    0
}
