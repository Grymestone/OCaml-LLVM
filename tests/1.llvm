; ModuleID = 'tests/1.llvm'
source_filename = "awesome jit"

@buf = global [4 x i8] c"%d\0A\00"
@buf.1 = global [4 x i8] c"%f\0A\00"

declare i32 @printf(i8*, ...)

; Function Attrs: noinline nounwind optnone sspstrong uwtable
define i32 @do_math(i32 %a) #0 {
entry:
  %multmp = mul i32 %a, 8
  ret i32 %multmp
}

; Function Attrs: noinline nounwind optnone sspstrong uwtable
define double @do_math2(double %b) #0 {
entry:
  %multmp = fmul double %b, 1.000000e+01
  ret double %multmp
}

; Function Attrs: noinline nounwind optnone sspstrong uwtable
define i32 @main() #0 {
entry:
  %calltmp = call double @do_math2(double 2.000000e+00)
  %calltmp1 = call i32 @do_math(i32 10)
  %0 = call i32 (i8*, ...) @printf(i8* getelementptr inbounds ([4 x i8], [4 x i8]* @buf, i32 0, i32 0), i32 %calltmp1)
  %1 = call i32 (i8*, ...) @printf(i8* getelementptr inbounds ([4 x i8], [4 x i8]* @buf.1, i32 0, i32 0), double %calltmp)
  ret i32 0
}

attributes #0 = { noinline nounwind optnone sspstrong uwtable }
