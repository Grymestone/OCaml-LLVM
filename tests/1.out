FuncDec: int do_math
	args:VarDec: int a

	code: VarDecExpr: int x = id(a)  *  5  +  3 

Expr methodcall do_math 10 

Declaring function do_math
Setting variable x
Operator *
Variable a
Operator +
 type LHS i32 type RHS i32
 type LHS i32 type RHS i32
Added return 

Calling function do_math

